thrift0.13/src/main/java/com/linecorp/armeria/server/thrift/THttpService.java

```diff
@@ -266,36 +258,25 @@ public static THttpService ofFormats(
     public static Function<? super RpcService, THttpService> newDecorator(
             SerializationFormat defaultSerializationFormat,
             Iterable<SerializationFormat> otherSupportedSerializationFormats) {
-
-        final SerializationFormat[] supportedSerializationFormatArray = newSupportedSerializationFormats(
-                defaultSerializationFormat, otherSupportedSerializationFormats);
-
-        return delegate -> new THttpService(delegate, supportedSerializationFormatArray);
-    }
-
-    private static SerializationFormat[] newSupportedSerializationFormats(
-            SerializationFormat defaultSerializationFormat,
-            Iterable<SerializationFormat> otherSupportedSerializationFormats) {
-
-        requireNonNull(defaultSerializationFormat, "defaultSerializationFormat");
-        requireNonNull(otherSupportedSerializationFormats, "otherSupportedSerializationFormats");
-
-        final Set<SerializationFormat> set = new LinkedHashSet<>();
-        set.add(defaultSerializationFormat);
-        Iterables.addAll(set, otherSupportedSerializationFormats);
-        return set.toArray(EMPTY_FORMATS);
+        return builder().defaultSerializationFormat(defaultSerializationFormat)
+                        .otherSerializationFormats(otherSupportedSerializationFormats)
+                        .newDecorator();
     }
 
-    private final SerializationFormat[] supportedSerializationFormatArray;
-    private final Set<SerializationFormat> supportedSerializationFormats;
     private final ThriftCallService thriftService;
+    private final SerializationFormat defaultSerializationFormat;
+    private final Set<SerializationFormat> supportedSerializationFormats;
+    private final BiFunction<ServiceRequestContext, ? super Throwable, ? extends Throwable> exceptionMapper;
```

Comment:
Like we did in others
https://github.com/line/armeria/blob/bd7fe49f6999af1d1c26b19dd7a220008c7ea202/core/src/main/java/com/linecorp/armeria/server/ServiceRequestContext.java#L106
```suggestion
    private final BiFunction<? super ServiceRequestContext, ? super Throwable, ? extends Throwable> exceptionMapper;
```

Reply:
Fixed. 😉